apps/studio/components/interfaces/SQLEditor/SQLEditor.tsx

```diff
@@ -414,7 +414,15 @@ const SQLEditor = () => {
         )
       }
     }
-  }, [debugSql, entityDefinitions, id, snap.results, snap.snippets])
+  }, [
+    debugSql,
+    entityDefinitions,
+    id,
+    snap.results,
+    snap.snippets,
+    snapV2.results,
+    snapV2.snippets,
```

Comment:
Hmm maybe results and snippets could be one variable here up top like

const results = enableFolders ? snapV2.results : snap.results

Likewise for snippets - so we reduce the amount of vars in the dependency array

Reply:
hummm I think we need the snapv2.results in the array?  if we cache them at the top of this function, they won't trigger a change right? Or I'm misunderstanding